### PR TITLE
Find largest AddressBook DB not just the first one that exists

### PR DESCRIPTION
**Pull request by alexdanilowicz**
Find the largest AddressBook DB, not just the first one that exists because we want the maximum number of contacts possible. 

Also, add a few stopwords and lowercase the list. 

Discovered this by testing this on a friend's chat.db and address book.

### CODE REVIEW

File: electron/src/db/index.ts [hunk 1]
```diff
@@ -14,31 +14,38 @@ export function closeDB(db: sqlite3.Database) {
...
-    return true;
+    return checkResult[0].count;
   }
   log.info(`${db}: No records found`);
-  return false;
+  return 0;
 }
 
-export async function returnDBIfPopulated(
+export interface DBRecordCount {
```
Comment (ayc19) [inline electron/src/db/index.ts:30]: DBWithRecordCount

Comment (ayc19) [reply electron/src/db/index.ts:30]: also why do we want record count

File: electron/src/db/index.ts [hunk 2]
```diff
@@ -14,31 +14,38 @@ export function closeDB(db: sqlite3.Database) {
...
   log.info(`Attempting to initialize ${dbPath}`);
   if (fs.existsSync(dbPath)) {
     const db = initializeDB(dbPath);
-    if (await checkIfRecordsExist(db, checkQuery)) {
-      log.info(`${db} populated`);
-      return db;
+    const recordCount = await getRecordCounts(db, checkQuery);
+    if (recordCount > 0) {
+      log.info(`${dbPath} populated`);
+      return { db, recordCount };
```
Comment (ayc19) [inline electron/src/db/index.ts:45]: is this return redundant?

File: electron/src/utils/index.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export function delimList(myList: string[]): string {
+  const l = myList.map((t) => `"${t}"`);
+  return l.join(', ');
+}
+
+export function lowerCaseList(myList: string[]): string[] {
+  const l = myList.map((t) => t.toLowerCase());
```
Comment (ayc19) [inline electron/src/utils/index.ts:7]: should just return myList.map(...)

File: electron/src/db/index.ts (hunk 1 above)
Comment (alexdanilowicz) [reply electron/src/db/index.ts:30]: we want record count to know the largest

File: electron/src/addressBro/util/index.ts [hunk 4]
```diff
@@ -20,16 +20,18 @@ async function readAddressBookBackups(): Promise<sqlite3.Database | undefined> {
...
+        return getDBWithRecordCounts(path, COUNT_CONTACTS_QUERY);
       })
     );
-    // NOTE: grab the first one that is populated and exit, instead of getting the biggest
-    const populatedDB = dbResults.find((p) => p);
-    if (populatedDB) {
-      return populatedDB;
+    if (dbRecordCountResults && dbRecordCountResults[0]) {
+      // get db with biggest record counts
+      const getMaxDBRC = (dbrcList: DBRecordCount[]) =>
```
Comment (ayc19) [inline electron/src/addressBro/util/index.ts:31]: Think you can just call the reduce directly

File: electron/src/utils/index.ts (hunk 3 above)
Comment (alexdanilowicz) [reply electron/src/utils/index.ts:7]: great callout thnx

File: electron/src/db/index.ts (hunk 2 above)
Comment (alexdanilowicz) [reply electron/src/db/index.ts:45]: Are you suggesting it is bc of the return on line 48?

I'll change the logic around so we dont have two returns. It's currently not redundant cause we wanna only closeDB if recordCount < 1

**Pull request merged**